牛马挣劳务队伍提交 patch 和 README

Hongzunhei opened the pull request:
队伍名：牛马挣劳务
赛题：多模态
提交文件：patch.zip、readme.md

Hongzunhei committed:
feat: add submission for team 牛马挣劳务

Hongzunhei committed:
refactor: move team folder to MultiModal

Hongzunhei committed:
Add files via upload

qhzhuang commented on 2025-Ascend-Innovation-Contest/S1/MultiModal/牛马挣劳务/README.md:
```diff
@@ -0,0 +1,77 @@
+   - **将VisionAttention当中的softmax去掉提升精度**:通过**flash_attention_score**融合算子也可以。
+## 三、总结
+改动较多，主要列举了收益较大的改动，其余改动见文件所示。未一一列举请多多包涵。
+
```
此处可以增加下最终得分 参考https://github.com/skyous779/competition/blob/f2eee6acee632e1aacf4990ed948528e9be0fdce/2025-Ascend-Innovation-Contest/S1/MoE/blueblue/README.md%E5%AE%8C%E5%96%84%E4%B8%8B

qhzhuang replied:
可以最后给出总的结果, 参考https://github.com/skyous779/competition/blob/f2eee6acee632e1aacf4990ed948528e9be0fdce/2025-Ascend-Innovation-Contest/S1/MoE/blueblue/README.md

qhzhuang commented on 2025-Ascend-Innovation-Contest/S1/MultiModal/牛马挣劳务/README.md:
````diff
@@ -0,0 +1,77 @@
+   - 在Qwen/Qwen2-VL-2B-Instruct当中，Prefill时延瓶颈主要来源于视觉部分。详细参考mindnlp/transformers/models/qwen2_vl/modeling_qwen2_vl.py。
+     通过Profiler工具可以观察到，有一个Conv3D的算子相当的耗时，将nn.Conv3d替换为mindspore.mint.nn.Conv3d能大大减少耗时，收益较大。
+   - 另外一部分瓶颈来源于**VisionAttention**的计算。通过**flash_attention_score**融合算子可以大大减少Prefill时延，收益较大。
+   ```python
````
感谢贡献的详细的优化思路说明, 在每个代码片段前可以增加下原实现作为对比, 更清晰.

qhzhuang commented on 2025-Ascend-Innovation-Contest/S1/MultiModal/牛马挣劳务/README.md:
```diff
@@ -0,0 +1,77 @@
+
+主要针对 **Prefill 时延、Decode 时延、峰值显存占用** 三方面的优化展开：
+
+1. **Prefill 时延**
```
建议使用二级标题

qhzhuang commented on 2025-Ascend-Innovation-Contest/S1/MultiModal/牛马挣劳务/README.md:
```diff
@@ -0,0 +1,77 @@
+
+   Prefill 时延主要是输入到产生第一个 token 的时间，可以看作「预处理 + 图像前向 + 文本编码」整体耗时。
+   
+    deepseek-ai/Janus-Pro-7B：
```
建议使用三级标题

qhzhuang commented on 2025-Ascend-Innovation-Contest/S1/MultiModal/牛马挣劳务/README.md:
```diff
@@ -0,0 +1,77 @@
+   - 在deepseek-ai/Janus-Pro-7B当中，Prefill时延瓶颈主要来源于预处理部分。详细参考llm/inference/janus_pro/janus/models/processing_vlm.py。
+     本人采用了一种一种笨笨的办法，通过[JIT装饰器](https://www.mindspore.cn/docs/zh-CN/r2.6.0/api_python/mindspore/mindspore.jit.html)加速操作。JIT装饰器主要作用是可以针对相同shape下的多个算子一起下发，节约时间。一定要是相同shape，否则将会重新编译，编译时长可能大于JIT能节约的时间。通过填充无用数据构造成相同shape后，再将无用数据剔除，还原原本shape的操作，可以将编译时间丢到warm-up里，后面正式进行推理时则节约了时间。具体收益不记得了，但是收益较大。
+
+     Qwen/Qwen2-VL-2B-Instruct：
```
这里缩进不对,建议使用三级标题

qhzhuang commented on 2025-Ascend-Innovation-Contest/S1/MultiModal/牛马挣劳务/README.md:
```diff
@@ -0,0 +1,77 @@
+# 队伍：牛马挣劳务
```
感谢给出的详细优化思路解读,建议在每个代码片段前给出原代码片段, 最为对比更清晰

Hongzunhei committed:
Optimize performance and memory usage in models

Optimized prefill and decode latencies, reduced peak memory usage, and improved performance metrics for Qwen/Qwen2-VL-2B-Instruct and deepseek-ai/Janus-Pro-7B models.
